AssertionError: parameters did not match comb_class for the rule

To recreate (but took me 10 tries):
```python
from tilings.tilescope import TileScope, TileScopePack

basis = '0213_1230'
pack = TileScopePack.insertion_row_and_col_placements(row_only=True).make_fusion(apply_first=True)
css = TileScope(basis, pack)
spec = css.auto_search()
[spec.count_objects_of_size(n) for n in range(20)]
```

Bugged spec: https://gist.github.com/jaypantone/4fd0d1d5623649b5f785f9f8a0d87388

Traceback:
```python
---------------------------------------------------------------------------
AssertionError                            Traceback (most recent call last)
<ipython-input-16-a3b22b65031f> in <module>
----> 1 [spec.count_objects_of_size(n) for n in range(20)]

<ipython-input-16-a3b22b65031f> in <listcomp>(.0)
----> 1 [spec.count_objects_of_size(n) for n in range(20)]

~/Dropbox/Research/Active/2017-44-ATRAP/repos/comb_spec_searcher/comb_spec_searcher/specification.py in count_objects_of_size(self, n, **parameters)
    350         limit = n * self.number_of_rules()
    351         with RecursionLimit(limit):
--> 352             return self.root_rule.count_objects_of_size(n, **parameters)
    353
    354     def generate_objects_of_size(

~/Dropbox/Research/Active/2017-44-ATRAP/repos/comb_spec_searcher/comb_spec_searcher/strategies/rule.py in count_objects_of_size(self, n, **parameters)
    378                 self.subrecs is not None
    379             ), "you must call the set_subrecs function first"
--> 380             res = self.constructor.get_recurrence(self.subrecs, n, **parameters)
    381             self.count_cache[key] = res
    382         #     # THE FOLLOWING CODE SNIPPET IS FOR DEBUGGING PURPOSES

~/Dropbox/Research/Active/2017-44-ATRAP/repos/comb_spec_searcher/comb_spec_searcher/strategies/constructor.py in get_recurrence(self, subrecs, n, **parameters)
    336     def get_recurrence(self, subrecs: SubRecs, n: int, **parameters: int) -> int:
    337         if not parameters:
--> 338             return sum(rec(n) for rec in subrecs)
    339         res = 0
    340         for (idx, rec), extra_params in zip(

~/Dropbox/Research/Active/2017-44-ATRAP/repos/comb_spec_searcher/comb_spec_searcher/strategies/constructor.py in <genexpr>(.0)
    336     def get_recurrence(self, subrecs: SubRecs, n: int, **parameters: int) -> int:
    337         if not parameters:
--> 338             return sum(rec(n) for rec in subrecs)
    339         res = 0
    340         for (idx, rec), extra_params in zip(

~/Dropbox/Research/Active/2017-44-ATRAP/repos/comb_spec_searcher/comb_spec_searcher/strategies/rule.py in count_objects_of_size(self, n, **parameters)
    378                 self.subrecs is not None
    379             ), "you must call the set_subrecs function first"
--> 380             res = self.constructor.get_recurrence(self.subrecs, n, **parameters)
    381             self.count_cache[key] = res
    382         #     # THE FOLLOWING CODE SNIPPET IS FOR DEBUGGING PURPOSES

~/Dropbox/Research/Active/2017-44-ATRAP/repos/comb_spec_searcher/comb_spec_searcher/strategies/constructor.py in get_recurrence(self, subrecs, n, **parameters)
    336     def get_recurrence(self, subrecs: SubRecs, n: int, **parameters: int) -> int:
    337         if not parameters:
--> 338             return sum(rec(n) for rec in subrecs)
    339         res = 0
    340         for (idx, rec), extra_params in zip(

~/Dropbox/Research/Active/2017-44-ATRAP/repos/comb_spec_searcher/comb_spec_searcher/strategies/constructor.py in <genexpr>(.0)
    336     def get_recurrence(self, subrecs: SubRecs, n: int, **parameters: int) -> int:
    337         if not parameters:
--> 338             return sum(rec(n) for rec in subrecs)
    339         res = 0
    340         for (idx, rec), extra_params in zip(

~/Dropbox/Research/Active/2017-44-ATRAP/repos/comb_spec_searcher/comb_spec_searcher/strategies/rule.py in count_objects_of_size(self, n, **parameters)
    378                 self.subrecs is not None
    379             ), "you must call the set_subrecs function first"
--> 380             res = self.constructor.get_recurrence(self.subrecs, n, **parameters)
    381             self.count_cache[key] = res
    382         #     # THE FOLLOWING CODE SNIPPET IS FOR DEBUGGING PURPOSES

~/Dropbox/Research/Active/2017-44-ATRAP/repos/comb_spec_searcher/comb_spec_searcher/strategies/constructor.py in get_recurrence(self, subrecs, n, **parameters)
    210             tmp = 1
    211             for (i, rec), extra_param in zip(enumerate(subrecs), extra_params):
--> 212                 tmp *= rec(n=comp[i], **extra_param)
    213                 if tmp == 0:
    214                     break

~/Dropbox/Research/Active/2017-44-ATRAP/repos/comb_spec_searcher/comb_spec_searcher/strategies/rule.py in count_objects_of_size(self, n, **parameters)
    378                 self.subrecs is not None
    379             ), "you must call the set_subrecs function first"
--> 380             res = self.constructor.get_recurrence(self.subrecs, n, **parameters)
    381             self.count_cache[key] = res
    382         #     # THE FOLLOWING CODE SNIPPET IS FOR DEBUGGING PURPOSES

~/Dropbox/Research/Active/2017-44-ATRAP/repos/comb_spec_searcher/comb_spec_searcher/strategies/constructor.py in get_recurrence(self, subrecs, n, **parameters)
    336     def get_recurrence(self, subrecs: SubRecs, n: int, **parameters: int) -> int:
    337         if not parameters:
--> 338             return sum(rec(n) for rec in subrecs)
    339         res = 0
    340         for (idx, rec), extra_params in zip(

~/Dropbox/Research/Active/2017-44-ATRAP/repos/comb_spec_searcher/comb_spec_searcher/strategies/constructor.py in <genexpr>(.0)
    336     def get_recurrence(self, subrecs: SubRecs, n: int, **parameters: int) -> int:
    337         if not parameters:
--> 338             return sum(rec(n) for rec in subrecs)
    339         res = 0
    340         for (idx, rec), extra_params in zip(

~/Dropbox/Research/Active/2017-44-ATRAP/repos/comb_spec_searcher/comb_spec_searcher/strategies/rule.py in count_objects_of_size(self, n, **parameters)
    378                 self.subrecs is not None
    379             ), "you must call the set_subrecs function first"
--> 380             res = self.constructor.get_recurrence(self.subrecs, n, **parameters)
    381             self.count_cache[key] = res
    382         #     # THE FOLLOWING CODE SNIPPET IS FOR DEBUGGING PURPOSES

~/Dropbox/Research/Active/2017-44-ATRAP/repos/comb_spec_searcher/comb_spec_searcher/strategies/constructor.py in get_recurrence(self, subrecs, n, **parameters)
    336     def get_recurrence(self, subrecs: SubRecs, n: int, **parameters: int) -> int:
    337         if not parameters:
--> 338             return sum(rec(n) for rec in subrecs)
    339         res = 0
    340         for (idx, rec), extra_params in zip(

~/Dropbox/Research/Active/2017-44-ATRAP/repos/comb_spec_searcher/comb_spec_searcher/strategies/constructor.py in <genexpr>(.0)
    336     def get_recurrence(self, subrecs: SubRecs, n: int, **parameters: int) -> int:
    337         if not parameters:
--> 338             return sum(rec(n) for rec in subrecs)
    339         res = 0
    340         for (idx, rec), extra_params in zip(

~/Dropbox/Research/Active/2017-44-ATRAP/repos/comb_spec_searcher/comb_spec_searcher/strategies/rule.py in count_objects_of_size(self, n, **parameters)
    378                 self.subrecs is not None
    379             ), "you must call the set_subrecs function first"
--> 380             res = self.constructor.get_recurrence(self.subrecs, n, **parameters)
    381             self.count_cache[key] = res
    382         #     # THE FOLLOWING CODE SNIPPET IS FOR DEBUGGING PURPOSES

~/Dropbox/Research/Active/2017-44-ATRAP/repos/comb_spec_searcher/comb_spec_searcher/strategies/constructor.py in get_recurrence(self, subrecs, n, **parameters)
    336     def get_recurrence(self, subrecs: SubRecs, n: int, **parameters: int) -> int:
    337         if not parameters:
--> 338             return sum(rec(n) for rec in subrecs)
    339         res = 0
    340         for (idx, rec), extra_params in zip(

~/Dropbox/Research/Active/2017-44-ATRAP/repos/comb_spec_searcher/comb_spec_searcher/strategies/constructor.py in <genexpr>(.0)
    336     def get_recurrence(self, subrecs: SubRecs, n: int, **parameters: int) -> int:
    337         if not parameters:
--> 338             return sum(rec(n) for rec in subrecs)
    339         res = 0
    340         for (idx, rec), extra_params in zip(

~/Dropbox/Research/Active/2017-44-ATRAP/repos/comb_spec_searcher/comb_spec_searcher/strategies/rule.py in count_objects_of_size(self, n, **parameters)
    378                 self.subrecs is not None
    379             ), "you must call the set_subrecs function first"
--> 380             res = self.constructor.get_recurrence(self.subrecs, n, **parameters)
    381             self.count_cache[key] = res
    382         #     # THE FOLLOWING CODE SNIPPET IS FOR DEBUGGING PURPOSES

~/Dropbox/Research/Active/2017-44-ATRAP/repos/comb_spec_searcher/comb_spec_searcher/strategies/constructor.py in get_recurrence(self, subrecs, n, **parameters)
    210             tmp = 1
    211             for (i, rec), extra_param in zip(enumerate(subrecs), extra_params):
--> 212                 tmp *= rec(n=comp[i], **extra_param)
    213                 if tmp == 0:
    214                     break

~/Dropbox/Research/Active/2017-44-ATRAP/repos/comb_spec_searcher/comb_spec_searcher/strategies/rule.py in count_objects_of_size(self, n, **parameters)
    378                 self.subrecs is not None
    379             ), "you must call the set_subrecs function first"
--> 380             res = self.constructor.get_recurrence(self.subrecs, n, **parameters)
    381             self.count_cache[key] = res
    382         #     # THE FOLLOWING CODE SNIPPET IS FOR DEBUGGING PURPOSES

~/Dropbox/Research/Active/2017-44-ATRAP/repos/comb_spec_searcher/comb_spec_searcher/strategies/constructor.py in get_recurrence(self, subrecs, n, **parameters)
    336     def get_recurrence(self, subrecs: SubRecs, n: int, **parameters: int) -> int:
    337         if not parameters:
--> 338             return sum(rec(n) for rec in subrecs)
    339         res = 0
    340         for (idx, rec), extra_params in zip(

~/Dropbox/Research/Active/2017-44-ATRAP/repos/comb_spec_searcher/comb_spec_searcher/strategies/constructor.py in <genexpr>(.0)
    336     def get_recurrence(self, subrecs: SubRecs, n: int, **parameters: int) -> int:
    337         if not parameters:
--> 338             return sum(rec(n) for rec in subrecs)
    339         res = 0
    340         for (idx, rec), extra_params in zip(

~/Dropbox/Research/Active/2017-44-ATRAP/repos/comb_spec_searcher/comb_spec_searcher/strategies/rule.py in count_objects_of_size(self, n, **parameters)
    378                 self.subrecs is not None
    379             ), "you must call the set_subrecs function first"
--> 380             res = self.constructor.get_recurrence(self.subrecs, n, **parameters)
    381             self.count_cache[key] = res
    382         #     # THE FOLLOWING CODE SNIPPET IS FOR DEBUGGING PURPOSES

~/Dropbox/Research/Active/2017-44-ATRAP/repos/Tilings/tilings/strategies/assumption_insertion.py in get_recurrence(self, subrecs, n, **parameters)
     61             for k, val in zip(self.new_parameters, values):
     62                 new_params[k] = val
---> 63             res += subrec(n, **new_params)
     64         return res
     65

~/Dropbox/Research/Active/2017-44-ATRAP/repos/comb_spec_searcher/comb_spec_searcher/strategies/rule.py in count_objects_of_size(self, n, **parameters)
    378                 self.subrecs is not None
    379             ), "you must call the set_subrecs function first"
--> 380             res = self.constructor.get_recurrence(self.subrecs, n, **parameters)
    381             self.count_cache[key] = res
    382         #     # THE FOLLOWING CODE SNIPPET IS FOR DEBUGGING PURPOSES

~/Dropbox/Research/Active/2017-44-ATRAP/repos/comb_spec_searcher/comb_spec_searcher/strategies/constructor.py in get_recurrence(self, subrecs, n, **parameters)
    347             ):
    348                 continue
--> 349             res += rec(n=n, **extra_params)
    350         return res
    351

~/Dropbox/Research/Active/2017-44-ATRAP/repos/comb_spec_searcher/comb_spec_searcher/strategies/rule.py in count_objects_of_size(self, n, **parameters)
    378                 self.subrecs is not None
    379             ), "you must call the set_subrecs function first"
--> 380             res = self.constructor.get_recurrence(self.subrecs, n, **parameters)
    381             self.count_cache[key] = res
    382         #     # THE FOLLOWING CODE SNIPPET IS FOR DEBUGGING PURPOSES

~/Dropbox/Research/Active/2017-44-ATRAP/repos/comb_spec_searcher/comb_spec_searcher/strategies/constructor.py in get_recurrence(self, subrecs, n, **parameters)
    347             ):
    348                 continue
--> 349             res += rec(n=n, **extra_params)
    350         return res
    351

~/Dropbox/Research/Active/2017-44-ATRAP/repos/comb_spec_searcher/comb_spec_searcher/strategies/rule.py in count_objects_of_size(self, n, **parameters)
    378                 self.subrecs is not None
    379             ), "you must call the set_subrecs function first"
--> 380             res = self.constructor.get_recurrence(self.subrecs, n, **parameters)
    381             self.count_cache[key] = res
    382         #     # THE FOLLOWING CODE SNIPPET IS FOR DEBUGGING PURPOSES

~/Dropbox/Research/Active/2017-44-ATRAP/repos/comb_spec_searcher/comb_spec_searcher/strategies/constructor.py in get_recurrence(self, subrecs, n, **parameters)
    210             tmp = 1
    211             for (i, rec), extra_param in zip(enumerate(subrecs), extra_params):
--> 212                 tmp *= rec(n=comp[i], **extra_param)
    213                 if tmp == 0:
    214                     break

~/Dropbox/Research/Active/2017-44-ATRAP/repos/comb_spec_searcher/comb_spec_searcher/strategies/rule.py in count_objects_of_size(self, n, **parameters)
    378                 self.subrecs is not None
    379             ), "you must call the set_subrecs function first"
--> 380             res = self.constructor.get_recurrence(self.subrecs, n, **parameters)
    381             self.count_cache[key] = res
    382         #     # THE FOLLOWING CODE SNIPPET IS FOR DEBUGGING PURPOSES

~/Dropbox/Research/Active/2017-44-ATRAP/repos/comb_spec_searcher/comb_spec_searcher/strategies/constructor.py in get_recurrence(self, subrecs, n, **parameters)
    347             ):
    348                 continue
--> 349             res += rec(n=n, **extra_params)
    350         return res
    351

~/Dropbox/Research/Active/2017-44-ATRAP/repos/comb_spec_searcher/comb_spec_searcher/strategies/rule.py in count_objects_of_size(self, n, **parameters)
    378                 self.subrecs is not None
    379             ), "you must call the set_subrecs function first"
--> 380             res = self.constructor.get_recurrence(self.subrecs, n, **parameters)
    381             self.count_cache[key] = res
    382         #     # THE FOLLOWING CODE SNIPPET IS FOR DEBUGGING PURPOSES

~/Dropbox/Research/Active/2017-44-ATRAP/repos/comb_spec_searcher/comb_spec_searcher/strategies/constructor.py in get_recurrence(self, subrecs, n, **parameters)
    347             ):
    348                 continue
--> 349             res += rec(n=n, **extra_params)
    350         return res
    351

~/Dropbox/Research/Active/2017-44-ATRAP/repos/comb_spec_searcher/comb_spec_searcher/strategies/rule.py in count_objects_of_size(self, n, **parameters)
    378                 self.subrecs is not None
    379             ), "you must call the set_subrecs function first"
--> 380             res = self.constructor.get_recurrence(self.subrecs, n, **parameters)
    381             self.count_cache[key] = res
    382         #     # THE FOLLOWING CODE SNIPPET IS FOR DEBUGGING PURPOSES

~/Dropbox/Research/Active/2017-44-ATRAP/repos/comb_spec_searcher/comb_spec_searcher/strategies/constructor.py in get_recurrence(self, subrecs, n, **parameters)
    347             ):
    348                 continue
--> 349             res += rec(n=n, **extra_params)
    350         return res
    351

~/Dropbox/Research/Active/2017-44-ATRAP/repos/comb_spec_searcher/comb_spec_searcher/strategies/rule.py in count_objects_of_size(self, n, **parameters)
    378                 self.subrecs is not None
    379             ), "you must call the set_subrecs function first"
--> 380             res = self.constructor.get_recurrence(self.subrecs, n, **parameters)
    381             self.count_cache[key] = res
    382         #     # THE FOLLOWING CODE SNIPPET IS FOR DEBUGGING PURPOSES

~/Dropbox/Research/Active/2017-44-ATRAP/repos/Tilings/tilings/strategies/assumption_splitting.py in get_recurrence(self, subrecs, n, **parameters)
     69         res = 0
     70         for sub_params in self._valid_compositions(**parameters):
---> 71             res += rec(n, **sub_params)
     72         return res
     73

~/Dropbox/Research/Active/2017-44-ATRAP/repos/comb_spec_searcher/comb_spec_searcher/strategies/rule.py in count_objects_of_size(self, n, **parameters)
    378                 self.subrecs is not None
    379             ), "you must call the set_subrecs function first"
--> 380             res = self.constructor.get_recurrence(self.subrecs, n, **parameters)
    381             self.count_cache[key] = res
    382         #     # THE FOLLOWING CODE SNIPPET IS FOR DEBUGGING PURPOSES

~/Dropbox/Research/Active/2017-44-ATRAP/repos/comb_spec_searcher/comb_spec_searcher/strategies/constructor.py in get_recurrence(self, subrecs, n, **parameters)
    210             tmp = 1
    211             for (i, rec), extra_param in zip(enumerate(subrecs), extra_params):
--> 212                 tmp *= rec(n=comp[i], **extra_param)
    213                 if tmp == 0:
    214                     break

~/Dropbox/Research/Active/2017-44-ATRAP/repos/comb_spec_searcher/comb_spec_searcher/strategies/rule.py in count_objects_of_size(self, n, **parameters)
    378                 self.subrecs is not None
    379             ), "you must call the set_subrecs function first"
--> 380             res = self.constructor.get_recurrence(self.subrecs, n, **parameters)
    381             self.count_cache[key] = res
    382         #     # THE FOLLOWING CODE SNIPPET IS FOR DEBUGGING PURPOSES

~/Dropbox/Research/Active/2017-44-ATRAP/repos/comb_spec_searcher/comb_spec_searcher/strategies/constructor.py in get_recurrence(self, subrecs, n, **parameters)
    347             ):
    348                 continue
--> 349             res += rec(n=n, **extra_params)
    350         return res
    351

~/Dropbox/Research/Active/2017-44-ATRAP/repos/comb_spec_searcher/comb_spec_searcher/strategies/rule.py in count_objects_of_size(self, n, **parameters)
    378                 self.subrecs is not None
    379             ), "you must call the set_subrecs function first"
--> 380             res = self.constructor.get_recurrence(self.subrecs, n, **parameters)
    381             self.count_cache[key] = res
    382         #     # THE FOLLOWING CODE SNIPPET IS FOR DEBUGGING PURPOSES

~/Dropbox/Research/Active/2017-44-ATRAP/repos/comb_spec_searcher/comb_spec_searcher/strategies/constructor.py in get_recurrence(self, subrecs, n, **parameters)
    336     def get_recurrence(self, subrecs: SubRecs, n: int, **parameters: int) -> int:
    337         if not parameters:
--> 338             return sum(rec(n) for rec in subrecs)
    339         res = 0
    340         for (idx, rec), extra_params in zip(

~/Dropbox/Research/Active/2017-44-ATRAP/repos/comb_spec_searcher/comb_spec_searcher/strategies/constructor.py in <genexpr>(.0)
    336     def get_recurrence(self, subrecs: SubRecs, n: int, **parameters: int) -> int:
    337         if not parameters:
--> 338             return sum(rec(n) for rec in subrecs)
    339         res = 0
    340         for (idx, rec), extra_params in zip(

~/Dropbox/Research/Active/2017-44-ATRAP/repos/comb_spec_searcher/comb_spec_searcher/strategies/rule.py in count_objects_of_size(self, n, **parameters)
    378                 self.subrecs is not None
    379             ), "you must call the set_subrecs function first"
--> 380             res = self.constructor.get_recurrence(self.subrecs, n, **parameters)
    381             self.count_cache[key] = res
    382         #     # THE FOLLOWING CODE SNIPPET IS FOR DEBUGGING PURPOSES

~/Dropbox/Research/Active/2017-44-ATRAP/repos/Tilings/tilings/strategies/assumption_insertion.py in get_recurrence(self, subrecs, n, **parameters)
     61             for k, val in zip(self.new_parameters, values):
     62                 new_params[k] = val
---> 63             res += subrec(n, **new_params)
     64         return res
     65

~/Dropbox/Research/Active/2017-44-ATRAP/repos/comb_spec_searcher/comb_spec_searcher/strategies/rule.py in count_objects_of_size(self, n, **parameters)
    378                 self.subrecs is not None
    379             ), "you must call the set_subrecs function first"
--> 380             res = self.constructor.get_recurrence(self.subrecs, n, **parameters)
    381             self.count_cache[key] = res
    382         #     # THE FOLLOWING CODE SNIPPET IS FOR DEBUGGING PURPOSES

~/Dropbox/Research/Active/2017-44-ATRAP/repos/comb_spec_searcher/comb_spec_searcher/strategies/constructor.py in get_recurrence(self, subrecs, n, **parameters)
    347             ):
    348                 continue
--> 349             res += rec(n=n, **extra_params)
    350         return res
    351

~/Dropbox/Research/Active/2017-44-ATRAP/repos/comb_spec_searcher/comb_spec_searcher/strategies/rule.py in count_objects_of_size(self, n, **parameters)
    370             "parameters did not match comb_class for the rule \n{}\nparameters "
    371             "given: {}\n comb_class_parameters: {}".format(
--> 372                 self, list(parameters), self.comb_class.extra_parameters,
    373             )
    374         )

AssertionError: parameters did not match comb_class for the rule
placing the topmost point in row 0
+-+-+                                    +-+                        +-+-+-+                                      +-+-+-+-+
|/|1|                                 =  |1|                     +  | |●| |                                   +  | | |●| |
+-+-+                                    +-+                        +-+-+-+                                      +-+-+-+-+
/: Av+(10)                               1: Av(ε)                   |/| |1|                                      |/|/| |1|
1: Av+(102, 1230)                        Crossing obstructions:     +-+-+-+                                      +-+-+-+-+
Crossing obstructions:                   ε:                         /: Av+(10)                                   /: Av+(10)
102: (0, 0), (1, 0), (1, 0)                                         1: Av+(102, 1230)                            /: Av(10)
120: (0, 0), (0, 0), (1, 0)                                         ●: point                                     1: Av(102, 1230)
1230: (0, 0), (1, 0), (1, 0), (1, 0)                                Crossing obstructions:                       ●: point
Requirement 0:                                                      10: (0, 0), (2, 0)                           Crossing obstructions:
0: (1, 0)                                                           Requirement 0:                               10: (0, 0), (1, 0)
Requirement 1:                                                      0: (0, 0)                                    102: (0, 0), (3, 0), (3, 0)
01: (0, 0), (0, 0)                                                  Requirement 1:                               102: (1, 0), (3, 0), (3, 0)
Assumption 0:                                                       0: (1, 1)                                    120: (0, 0), (0, 0), (3, 0)
can count points in cell (0, 0)                                     Requirement 2:                               120: (0, 0), (1, 0), (3, 0)
                                                                    0: (2, 0)                                    120: (1, 0), (1, 0), (3, 0)
                                                                    Assumption 0:                                1230: (0, 0), (3, 0), (3, 0), (3, 0)
                                                                    can count points in cells (0, 0), (1, 1)     1230: (1, 0), (3, 0), (3, 0), (3, 0)
                                                                                                                 Requirement 0:
                                                                                                                 0: (2, 1)
                                                                                                                 Requirement 1:
                                                                                                                 01: (0, 0), (0, 0)
                                                                                                                 Assumption 0:
                                                                                                                 can count points in cell (0, 0)
parameters given: []
 comb_class_parameters: ('k_0',)
```

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -20,8 +20,7 @@ and this project adheres to [Semantic Versioning](http://semver.org/spec/v2.0.0.
   child, preventing a `KeyError`.
 - the extra parameters dictionary is flipped when creating the constructor in a
   reverse rule.
-- fixed the extra parameter map in `EquivalencePathRule` which was not
-  composing correctly.
+- fixed the `EquivalencePathRule.constructor` method
 
 ### Changed
 - When the searcher finds a specification, it will now spends 1% of the time

--- a/comb_spec_searcher/strategies/rule.py
+++ b/comb_spec_searcher/strategies/rule.py
@@ -480,22 +480,25 @@ class EquivalenceRule(Rule[CombinatorialClassType, CombinatorialObjectType]):
         super().__init__(rule.strategy, rule.comb_class, (child,))
         self.child_idx = rule.children.index(child)
         self.actual_children = rule.children
-        self._constructor: Optional[Constructor] = None
+        self._constructor: Optional[DisjointUnion] = None
 
     @property
-    def constructor(
-        self,
-    ) -> Constructor[CombinatorialClassType, CombinatorialObjectType]:
+    def constructor(self,) -> DisjointUnion:
         """
         Return the constructor, that contains all the information about how to
         count/generate objects from the rule.
         """
         if self._constructor is None:
-            self._constructor = self.strategy.constructor(
-                self.comb_class, self.actual_children
+            original_constructor = cast(
+                DisjointUnion,
+                self.strategy.constructor(self.comb_class, self.actual_children),
             )
-            if self._constructor is None:
-                raise StrategyDoesNotApply("{} does not apply".format(self.strategy))
+            assert isinstance(original_constructor, DisjointUnion)
+            self._constructor = DisjointUnion(
+                self.comb_class,
+                self.children,
+                (original_constructor.extra_parameters[self.child_idx],),
+            )
         return self._constructor
 
     @property
@@ -548,22 +551,9 @@ class EquivalencePathRule(Rule[CombinatorialClassType, CombinatorialObjectType])
                 k: k for k in self.comb_class.extra_parameters
             }
             for rule in self.rules:
-                # TODO: call extra parameters on Rule class
-                if isinstance(rule, EquivalenceRule):
-                    rules_parameters = rule.strategy.extra_parameters(
-                        rule.comb_class, rule.actual_children
-                    )[rule.child_idx]
-                elif isinstance(rule, ReverseRule):
-                    forward_rule_parameters = rule.strategy.extra_parameters(
-                        rule.children[0]
-                    )[0]
-                    rules_parameters = {
-                        b: a for a, b in forward_rule_parameters.items()
-                    }
-                else:
-                    rules_parameters = rule.strategy.extra_parameters(
-                        rule.comb_class, rule.children
-                    )[0]
+                original_constructor = rule.constructor
+                assert isinstance(original_constructor, DisjointUnion)
+                rules_parameters = original_constructor.extra_parameters[0]
                 extra_parameters = {
                     parent_var: rules_parameters[child_var]
                     for parent_var, child_var in extra_parameters.items()
@@ -660,13 +650,13 @@ class ReverseRule(Rule[CombinatorialClassType, CombinatorialObjectType]):
     @property
     def constructor(self) -> DisjointUnion:
         if self._constructor is None:
-            constructor = cast(DisjointUnion, self.original_rule.constructor)
-            assert isinstance(constructor, DisjointUnion), (
+            original_constructor = self.original_rule.constructor
+            assert isinstance(original_constructor, DisjointUnion), (
                 "reverse rule coming from non disjoint union strategy - "
                 "you'll need to update the ReverseRule constructor!"
             )
             flipped_extra_params = {
-                b: a for a, b in constructor.extra_parameters[0].items()
+                b: a for a, b in original_constructor.extra_parameters[0].items()
             }
             self._constructor = DisjointUnion(
                 self.comb_class, self.children, (flipped_extra_params,)

